lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1294,6 +1295,23 @@ def run(self):
 
         return msg, changed, rebuilt, differences.get_legacy_docker_diffs(), facts
 
+    def run_safe(self):
+        try:
+            return self.run()
+        except APIError as e:
+            # Sometimes Version.Index will have changed between an inspect and
+            # update. If this is encountered we'll retry the update.
+            if (
+                self.retries > 0
+                and e.explanation
+                and 'update out of sequence' in str(e.explanation)
+            ):
+                self.retries -= 1
+                time.sleep(1)
+                return self.run_safe()
```

Comment:
Why not use a loop instead of using tail recursion?

Reply:
At the moment I found it more readable and have less chances of paving way for introducing bug which would iterate over `run()` unintentionally. But you're right its more of a best practice to use a loop for retrying, especially when checking how other modules do it. I'll fix it.